libs/types/src/smartcard_auth/dipped_smartcard_auth.ts

```diff
@@ -14,7 +14,7 @@ export interface CheckingPasscode {
   readonly status: 'checking_passcode';
   readonly user: AdminUser;
   readonly checkPasscode: (passcode: string) => void;
-  readonly passcodeError?: 'wrong_passcode';
+  readonly wrongPasscodeEntered?: true;
```

Comment:
Nice! I like this shift - I can't really imagine any passcode errors other than an incorrect passcode